specs/core/1_shard-data-chains.md

```diff
@@ -300,76 +302,52 @@ def is_valid_shard_block(beacon_blocks: Sequence[BeaconBlock],
             return True
 
     # Check slot number
-    assert candidate.slot >= PHASE_1_FORK_SLOT
-
-    # Check shard number
-    assert candidate.shard <= SHARD_COUNT
+    assert candidate.core.slot >= PHASE_1_FORK_SLOT
 
     # Check beacon block
-    beacon_block = beacon_blocks[candidate.slot]
-    assert candidate.beacon_block_root == signing_root(beacon_block)
-    assert beacon_block.slot <= candidate.slot
+    beacon_block = beacon_blocks[candidate.core.slot - candidate.core.slot % (SHARD_SLOTS_PER_BEACON_SLOT * SLOTS_PER_EPOCH)]
+    assert candidate.core.beacon_block_root == signing_root(beacon_block)
+    assert beacon_block.slot <= candidate.core.slot
 
     # Check state root
-    assert candidate.state_root == Hash()  # [to be removed in phase 2]
+    assert candidate.core.state_root == Hash()  # [to be removed in phase 2]
 
     # Check parent block
-    if candidate.slot == PHASE_1_FORK_SLOT:
-        assert candidate.parent_root == Hash()
+    if candidate.core.slot == PHASE_1_FORK_SLOT:
```

Comment:
```suggestion
    if (
        candidate.core.slot % SHARD_SLOTS_PER_BEACON_SLOT == 0
        and compute_slot_from_shard_slot(candidate.core.slot) == PHASE_1_FORK_SLOT:
    )
```

Reply:
Not sure this is actually correct. If the 0th slot of a shard chain is skipped, we need a way to recover

Reply:
Seems more like the conditional should be `if candidate.core.parent_root == Hash(): do nothing` rather than checking the slot